chain/consensus/filcns/upgrades.go

```diff
@@ -2770,11 +2779,21 @@ func UpgradeActorsV16(ctx context.Context, sm *stmgr.StateManager, cache stmgr.M
 	if workerCount <= 0 {
 		workerCount = 1
 	}
+	timeout := os.Getenv("LOTUS_MIGRATE_PROGRESS_LOG_SECONDS")
+	timeoutDuration, err := time.ParseDuration(timeout + "s")
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("error parsing LOTUS_MIGRATE_PROGRESS_LOG_SECONDS: %w", err)
+	}
+
+	if timeoutDuration == 0 {
+		timeoutDuration = time.Second * 2
+	}
```

Comment:
```suggestion
	timeoutDuration := time.Second * 2
	timeout := os.Getenv("LOTUS_MIGRATE_PROGRESS_LOG_SECONDS")
	if timeout != "" {
		parsed, err := time.ParseDuration(timeout + "s")
		if err != nil {
			return cid.Undef, xerrors.Errorf("error parsing LOTUS_MIGRATE_PROGRESS_LOG_SECONDS: %w", err)
		}
		timeoutDuration = parsed
	}
```

You need to introduce a helper function to do this, you've done it twice now and on the next network upgrade it's going to be copied twice more, and so on. So let's have a helper function do it once.

Reply:
And .. I think `ParseDuration` is the wrong choice here - we're saying "SECONDS" in the env var, and you're appending an `s` to it. It could take non-int input and still be valid, I think we want to avoid confusion and just parse it straight as an int:

```suggestion
timeoutDuration := time.Second * 2
timeout := os.Getenv("LOTUS_MIGRATE_PROGRESS_LOG_SECONDS")
if timeout != "" {
    seconds, err := strconv.Atoi(timeout)
    if err != nil {
        return cid.Undef,  xerrors.Errorf("LOTUS_MIGRATE_PROGRESS_LOG_SECONDS must be an integer: %w", err)
    }
    if seconds <= 0 {
        return cid.Undef,  xerrors.Errorf("LOTUS_MIGRATE_PROGRESS_LOG_SECONDS must be positive")
    }
    timeoutDuration = time.Duration(seconds) * time.Second
}
```